tests/Doctrine/Tests/Common/Annotations/DocParserTest.php

```diff
@@ -422,6 +422,8 @@ public function getAnnotationVarTypeProviderValid()
               // array type
             array('array', '{@AnnotationExtendsAnnotationTargetAll}'),
             array('array', '{@AnnotationExtendsAnnotationTargetAll,@AnnotationExtendsAnnotationTargetAll}'),
+            array('array', '[@AnnotationExtendsAnnotationTargetAll]'),
+            array('array', '[@AnnotationExtendsAnnotationTargetAll,@AnnotationExtendsAnnotationTargetAll]'),
```

Comment:
Would need a few tests around parsing array elements, their order, and different types of sub-elements